sync/syncer.go

```diff
@@ -375,6 +394,21 @@ func (s *Syncer) ATXs(mb *types.Block) (atxs []*types.ActivationTx, associated *
 	return atxs, associated, nil
 }
 
+func (s *Syncer) EnsurePoetProofAvailableAndValid(poetProofRef []byte) error {
```

Comment:
PoetProofAvailable could be something like "Exist" you dot have to include "poet" in the name
it is implied by the var name  poetDb :) 
"EnsurePoetProofAvailableAndValid" should be something like FetchPoetProof or SyncPoetProof